### PR TITLE
change python to use a version stream and add python 3.10

### PR DESCRIPTION
**Pull request by rawlingsj**
Signed-off-by: James Rawlings <jrawlings@chainguard.dev>
Signed-off-by: Eric Timmons <etimmons@asylonrobotics.com>

### CODE REVIEW

File: python-3.11.yaml [hunk 1]
```diff
@@ -11,7 +11,8 @@ package:
       attestation: TODO
       license: PSF-2.0
   dependencies:
-    runtime:
+    provides:
+      - python=3.11
```
Comment (kaniini) [inline python-3.11.yaml:15]:
```suggestion
      - python3=3.11.999
```

File: python-3.11.yaml [hunk 2]
```diff
@@ -95,6 +96,8 @@ subpackages:
     dependencies:
       runtime:
         - python3
+      provides:
+        - python-dev=3.11
```
Comment (kaniini) [inline python-3.11.yaml:100]:
```suggestion
        - python3-dev=3.11.999
```

File: python-3.10.yaml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    runtime:
+
+secfixes:
+  3.10.9-r0:
+    - CVE-2020-10735
+
+environment:
+  contents:
+    repositories:
+      - https://packages.wolfi.dev/bootstrap/stage3
```
Comment (kaniini) [inline python-3.10.yaml:23]: I would move python-3.10 down in the build order and not depend on bootstrap for it.

**Review by kaniini — CHANGES_REQUESTED** [review]

**Review by kaniini — APPROVED** [review]

**Review by dlorenc — APPROVED** [review]

File: python-3.10.yaml [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  epoch: 0
+  description: "the Python programming language"
+  target-architecture:
+    - all
+  copyright:
+    - paths:
+        - "*"
+      attestation: TODO
+      license: PSF-2.0
+  dependencies:
```
Comment (imjasonh) [inline python-3.10.yaml:13]: Should this also `provides: python=3.10.999` ?

(I have the same question about the various Go's and Ruby's etc)

File: python-3.10.yaml (hunk 3 above)
Comment (imjasonh) [reply python-3.10.yaml:23]: This comment sounds like we shouldn't be depending on bootstrapping, am I misunderstanding the question?

**Comment by imjasonh** [comment]
This seems to have broken the build:

```
Feb  7 21:05:42.150 [DEBUG] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:] ERROR: unable to select packages:
Feb  7 21:05:42.151 [DEBUG] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:]   python3-dev-3.11.1-r4:
Feb  7 21:05:42.154 [DEBUG] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:]     conflicts: python3-dev-3.11.1-r4[python3-dev=3.11.999]
Feb  7 21:05:42.154 [DEBUG] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:]     satisfies: world[python3-dev]
```

https://github.com/wolfi-dev/os/actions/runs/4118151884/jobs/7110333106#step:12:13012

**Pull request merged**